### ISSUE TITLE
Microdeposit flow not working in 3.6.0 - opt-in

### ISSUE THREAD

**Issue by iamjosephmj** [plaid/plaid-link-android#195]
# The problem

**cannot verify the micro-deposit flow in the latest SDK** 

_why should we allow the user to type anything other than the amount to do the microdeposit verification in the 
text view (for the microdeposit verification flow)? 
You can see this in the video that I had attached below._

# Environment

Android OS Version                 : 9,10,11 
Android Devices/Emulators     : All devices and emulators.  

# Steps to Reproduce

Do a simple micro deposit verification flow.

# Expected Result
User should be a able to verify the microdeposit flow.

# Screenshots

https://user-images.githubusercontent.com/18631114/135067907-eaad941d-a4cc-4d5a-98e0-392fa5386146.mp4

# Logs
**This is the link response that we get from the SDK**
_2021-09-28 15:34:03.404 16879-16879/<package-name> E/Event: LinkEvent(eventName=ERROR, metadata=LinkEventMetadata(brandName=, errorCode=INTERNAL_SERVER_ERROR, errorMessage=an unexpected error occurred, errorType=API_ERROR, exitStatus=, institutionId=, institutionName=, institutionSearchQuery=, linkSessionId=dec7d293-dfef-42fa-a467-483f1e7d9fa6, mfaType=, requestId=7RBMIVbsNFjQh0E, selection=, timestamp=, viewName=null, metadataJson=null))_

**Comment by iamjosephmj** [comment]
@amytang0 do you need any details about the integration part?

**Comment by amytang0** [comment]
Heya, taking a look at this now, thank you for posting the link_session_id!

**Comment by amytang0** [comment]
Hi, would you mind typing in "$0.01" instead of "0.01"? I think our verification code might be a little too strict at the moment (in the process of fixing that now), but I'd like to see if the rest of the flow works. Thank you!

**Comment by iamjosephmj** [comment]
@amytang0 sure, let me give it a try...

**Comment by iamjosephmj** [comment]
@amytang0 the first deposit is working now..
This is how it looks. But the second one is failing..
https://user-images.githubusercontent.com/18631114/135148002-c6ae2742-cfb4-4e33-87b7-82b5f22bf80c.mp4

**Comment by iamjosephmj** [comment]
@amytang0 These are the logs for the same:

LinkEvent(eventName=TRANSITION_VIEW, metadata=LinkEventMetadata(brandName=, errorCode=, errorMessage=, errorType=, exitStatus=, institutionId=, institutionName=, institutionSearchQuery=, linkSessionId=c764cf60-58f1-4bd8-aba2-88928ed4b42b, mfaType=, requestId=UFjiflrrdCmMYs4, selection=, timestamp=, viewName=null, metadataJson=null))

LinkEvent(eventName=TRANSITION_VIEW, metadata=LinkEventMetadata(brandName=, errorCode=, errorMessage=, errorType=, exitStatus=, institutionId=, institutionName=, institutionSearchQuery=, linkSessionId=c764cf60-58f1-4bd8-aba2-88928ed4b42b, mfaType=, requestId=UBXfnYOYqMXTVc6, selection=, timestamp=, viewName=com.plaid.link.event.LinkEventViewName$ERROR@a94ee6e, metadataJson=null))

LinkEvent(eventName=ERROR, metadata=LinkEventMetadata(brandName=, errorCode=, errorMessage=, errorType=, exitStatus=, institutionId=, institutionName=, institutionSearchQuery=, linkSessionId=c764cf60-58f1-4bd8-aba2-88928ed4b42b, mfaType=, requestId=UBXfnYOYqMXTVc6, selection=, timestamp=, viewName=null, metadataJson=null))

I hope this helps..

**Comment by amytang0** [comment]
Absolutely, thank you so much! I'm landing a fix that should deploy by EOD, I will keep you posted

**Comment by amytang0** [comment]
Hi sorry, the deploy was blocked, but the fix is merged in. Fingers crossed we can get the lock unlocked by EOD

**Comment by iamjosephmj** [comment]
@amytang0 Sure

**Comment by iamjosephmj** [comment]
@amytang0 will it take 2+ days for the deployment? It will be great if you can give a heads-up on the release..

**Comment by amytang0** [comment]
Heya, the deploy is out! Sorry for not posting about it; just confirmed myself that the microdeposits flow works (and additionally should handle inputs without the dollar prefix)

**Comment by iamjosephmj** [comment]
@amytang0 do you need to enable the opt-in for 3.6.1rc1 for the native look and feel ?

**Comment by iamjosephmj** [comment]
@amytang0 do I need to use 3.6.1 rc1? If so, do I need to send you my client ID for activating opt-in ?

**Comment by amytang0** [comment]
No, this was a backend fix, so this should work with 3.6.0

**Comment by iamjosephmj** [comment]
@amytang0 The second microdeposit is not working for me, can you give me the amount that you had entered for the second microdeposit?

**Comment by amytang0** [comment]
0.02

**Comment by iamjosephmj** [comment]
@amytang0 I had tried it with 0.02.. But, It still show the same error 😔. I just saw that you had changed the status of this issue to "bug". It will be great if you can give a rough idea on when this issue will be resolved, so that I can notify our customers the same .. Sorry to keep you messaging on this..😕

**Comment by amytang0** [comment]
Apologies for the late reply, is this still repro-ing?
